src/sys/system.h

```diff
@@ -152,10 +162,23 @@ class System
      */
     const Config& GetConfig() const { return cfg_; }
 
-    /** Returns an enum representing the current (primary) memory space used
+    /** Returns an enum representing the current (primary) memory space used 
      *  for executing the program.
+     *  \param program_start Optionally indicate a specific address to be checked. 
+     *  Checks the VTOR by default, which will generally indicate the memory
+     *  from which the program is executed.
      */
-    static ProgramMemory GetProgramMemory();
+    static ProgramMemory GetProgramMemory(uint32_t program_start = SCB->VTOR);
+
+    static constexpr uint32_t sram_start = 0x24000000U;
+    static constexpr uint32_t sram_end   = sram_start + 0x80000U;
+    static constexpr uint32_t qspi_start = 0x90040000U;
+    // TODO -- this is a bit too large:
+    static constexpr uint32_t qspi_end       = qspi_start + 0x800000U;
+    static constexpr uint32_t internal_start = 0x08000000U;
+    static constexpr uint32_t internal_end   = internal_start + 0x20000U;
+
+    static constexpr uint32_t expected_stack = 0x20020000U;
```

Comment:
constant naming conventions should be:
```c++
static constexpr uint32_t kSramStart = 0x24000000U;
```
etc. to make clear that it's constant while reading.